web3/contract.py

```diff
@@ -808,14 +826,23 @@ def __getattr__(self, attr: Any) -> Callable[[], None]:
         return NonExistentFallbackFunction._raise_exception
 
 
+class NonExistentReceiveFunction:
+    @staticmethod
+    def _raise_exception() -> NoReturn:
+        raise FallbackNotFound("No receive function was found in the contract ABI.")
+
+    def __getattr__(self, attr: Any) -> Callable[[], None]:
+        return NonExistentReceiveFunction._raise_exception
+
+
 class ContractFunction:
     """Base class for contract functions
 
     A function accessed via the api contract.functions.myMethod(*args, **kwargs)
     is a subclass of this class.
     """
     address: ChecksumAddress = None
-    function_identifier: Union[str, Type[FallbackFn]] = None
+    function_identifier: Union[str, Type[FallbackFn], Type[ReceiveFn]] = None
```

Comment:
You might consider aliasing this:

```python
# web3/typing.py

FunctionIdentifier = Union[str, Type[FallbackFn], Type[ReceiveFn]]
```

Then you can reuse it across the codebase.

Reply:
Good idea!